numpy/core/src/multiarray/ctors.c

```diff
@@ -4022,6 +4021,14 @@ PyArray_FromIter(PyObject *obj, PyArray_Descr *dtype, npy_intp count)
             goto done;
         }
         ((PyArrayObject_fields *)ret)->data = new_data;
+
+        if (count == -1) {
```

Comment:
Possibly safer just to this unconditionally, in case the `count` logic changes later. It might even be faster to an irrelevant degree.

Reply:
Done

Reply:
Just to note, I kept this (added an `|| RELAXED_STRIDES_DEBUG` because I thought it might be necessary, but I guess it probably is not; remove again?).

The reason is, that the actual resetting needs to reset all strides in case of subarrays, so the whole thing is a bit more complicated.